test/shared/src/main/scala/zio/test/SummaryBuilder.scala

```diff
@@ -16,7 +16,7 @@ object SummaryBuilder {
                  case _                          => false
                }
       failures <- extractFailures(executedSpec)
-      rendered <- ZIO.foreach(failures)(DefaultTestReporter.render(_, TestAnnotationRenderer.silent))
+      rendered <- ZIO.foreach(failures)(renderer.render(_, TestAnnotationRenderer.silent))
```

Comment:
This previously coupled the `SummaryBuilder` to the `DefaultTestReporter`. I now pass the renderer from the outside as an argument, but wonder if passing it via `R` can be a better idea? WDYT?

Reply:
I think generally passing it as a parameter is simpler if we don't need the power of the environment to avoid plumbing the dependency through a bunch of different levels. Seems pretty straightforward here.